sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java

```diff
@@ -72,109 +73,119 @@ public class NodeService {
+    /**
+     * Add a node to the task of a defined job
+     * @param node a Iaas node definition
+     * @param job the job
+     */
+    @Transactional
+    public void addNode(IaasDefinition node, Job job) {
+        LOGGER.info("Adding IAAS node {} to job [{}]", node.toString(), job.getJobId());
+        Deployment newDeployment = new Deployment();
+        newDeployment.setNodeName(node.getName());
+        newDeployment.setDeploymentType(NodeType.IAAS);
 
-            LOGGER.info("Trying to retrieve task: " + node.getTaskName());
-            Task task = repositoryService.getJob(jobId).findTask(node.getTaskName());
-            if (task == null) {
-                LOGGER.error("Task [{}] does not exist in job [{}].", node.getTaskName(), jobId);
-                throw new IllegalArgumentException(String.format("node.taskName [%s] not valid.", node.getTaskName()));
-            }
+        LOGGER.info("Trying to retrieve node candidate and related iaas node: " + node.getNodeCandidateId());
+        NodeCandidate nodeCandidate = repositoryService.getNodeCandidate(node.getNodeCandidateId());
+        if (nodeCandidate == null) {
+            LOGGER.error("Node candidate [{}] does not exist.", node.getNodeCandidateId());
+            throw new IllegalArgumentException(String.format("node.nodeCandidateID [%s] not valid.",
+                                                             node.getNodeCandidateId()));
+        }
+        IaasNode iaasNode = repositoryService.getIaasNode(nodeCandidate.getNodeId());
 
-            newDeployment.setPaCloud(cloud);
-            newDeployment.setTask(task);
-            newDeployment.setNumber(task.getNextDeploymentID());
-            repositoryService.saveDeployment(newDeployment);
-            LOGGER.debug("Deployment created: " + newDeployment.toString());
+        newDeployment.setIaasNode(iaasNode);
+        iaasNode.incDeployedNodes(1L);
+        repositoryService.saveIaasNode(iaasNode);
 
-            repositoryService.savePACloud(cloud);
-            LOGGER.info("Deployment added to the related cloud: " + cloud.toString());
+        PACloud cloud = repositoryService.getPACloud(node.getCloudId());
+        if (cloud == null) {
+            LOGGER.error("Cloud [{}] does not exist.", node.getCloudId());
+            throw new IllegalArgumentException(String.format("node.cloudId [%s] not valid.", node.getCloudId()));
+        }
+        cloud.addDeployment(newDeployment);
+        if (WhiteListedInstanceTypesUtils.isHandledHardwareInstanceType(newDeployment.getNode()
+                                                                                     .getNodeCandidate()
+                                                                                     .getHardware()
+                                                                                     .getProviderId())) {
+            if (Boolean.FALSE.equals(cloud.isWhiteListedRegionDeployed(newDeployment.getNode()
```

Comment:
can't you use autoboxing ?
e.g.
```
if (!cloud.isWhiteListedRegionDeployed(newDeployment.getNode()
                                                                                    .getNodeCandidate()
                                                                                    .getLocation()
                                                                                    .getName()))
```

Reply:
This is to shut-up the sonar smell regarding booleans testing condition.
See [https://stackoverflow.com/questions/59035202/sonarlint-use-the-primitive-boolean-expression-here](here)

Reply:
ok if there can be null ...